### ISSUE TITLE
Unable to load X9.42 DH Parameters using PEM_read_bio_DHparams (3.0.0-master)

### ISSUE THREAD

**Issue by reaperhulk** [openssl/openssl#13046]
Fails on master (compiled today), passes on 1.1.1h. I haven't checked to see if this is failing in older alphas.

```c
#include <stdio.h>
#include <string.h>
#include <openssl/bio.h>
#include <openssl/dh.h>
#include <openssl/pem.h>

int main() {
    char *params = "-----BEGIN X9.42 DH PARAMETERS-----\nMIICKQKCAQEArRB+HpEjqdDWYPqnlVnFH6INZOVoO5/RtUsVl7YdCnXm+hQd+VpW\n26+aPEB7od8V6z1oijCcGA4d5rhaEnSgpm0/gVKtasISkDfJ7e/aTfjZHo/vVbc5\nS3rVt9C2wSIHyfmNEe002/bGugssi7wnvmoA4KC5xJcIs7+KMXCRiDaBKGEwvImF\n2xYC5xRBXZMwJ4Jzx94x79xzEPcSH9WgdBWYfZrcCkhtzfk6zEQyg4cxXXXhmMZB\npIDNhqG55YfovmDmnMkosrnFIXLkEwQumyPxCw4W55djybU9z0uoCinj+3PBa451\nuX7zY+L/ox9xz53lOE5xuBwKxN/+DBDmTwKCAQEArEAy708tmuOd8wtcj/2sUGze\nvnuJmYyvdIZqCM/k/+OmgkpOELmm8N2SHwGnDEr6q3OddwDCn1LFfbF8YgqGUr5e\nkAGo1mrXwXZpEBmZAkr00CcnWsE0i7inYtBSG8mK4kcVBCLqHtQJk51U2nRgzbX2\nxrJQcXy+8YDrNBGOmNEZUppF1vg0Vm4wJeMWozDvu3eobwwasVsFGuPUKMj4rLcK\ngTcVC47rEOGD7dGZY93Z4mPkdwWJ72qiHn9fL/OBtTnM40CdE81Wavu0jWwBkYHh\nvP6UswJp7f5y/ptqpL17Wg8ccc//TBnEGOH27AF5gbwIfypwZbOEuJDTGR8r+gId\nAIAcDTTFjZP+mXF3EB+AU1pHOM68vziambNjces=\n-----END X9.42 DH PARAMETERS-----\n";
    BIO *bio = BIO_new_mem_buf(params, strlen(params));
    DH *dh = PEM_read_bio_DHparams(bio, NULL, NULL, NULL);
    if (dh == NULL) {
        printf("failed to load\n");
        return 1;
    }
    return 0;
}
```

**Comment by kroeckx** [comment]
#13050 might be relevant

**Comment by reaperhulk** [comment]
This is still true as of alpha8.

**Comment by mattcaswell** [comment]
Found it. Another instance of us confusing DSA params and X9.42 params (because they have the same DER structure), because we forgot what the PEM header said. Fix in #13329. Fix is WIP because it lacks a test.

**Comment by mattcaswell** [comment]
This has been partially fixed by 35426b2. I say partially because the symptoms are resolved (the reproducer above no longer fails). The underlying bug in OSSL_STORE remains. See https://github.com/openssl/openssl/pull/13329#issuecomment-729650793 for details.
